clcache.py

```diff
@@ -294,6 +294,9 @@ def __init__(self, cacheDirectory=None):
         timeoutMs = int(os.environ.get('CLCACHE_OBJECT_CACHE_TIMEOUT_MS', 10 * 1000))
         self.lock = CacheLock(lockName, timeoutMs)
 
+    def openStatistics(self):
+        return Statistics(os.path.join(self.dir, "stats.txt"))
```

Comment:
About being a bit more funky (may I say 'pythonic'?) here and define a property which returns a context manager? Like:

```python
@property
def statistics(self):
    return Statistics(os.path.join(self.dir, "stats.txt"))
```

This allows writing

```python
with cache.statistics as stats:
    stats.doThis()
```